R/tm_a_regression.R

```diff
@@ -1036,25 +1048,71 @@ srv_a_regression <- function(id,
     )
 
     ### REPORTER
-    if (with_reporter) {
-      card_fun <- function(comment, label) {
-        card <- teal::report_card_template(
-          title = "Linear Regression Plot",
-          label = label,
-          with_filter = with_filter,
-          filter_panel_api = filter_panel_api
-        )
-        card$append_text("Plot", "header3")
-        card$append_plot(plot_r(), dim = pws$dim())
-        if (!comment == "") {
-          card$append_text("Comment", "header3")
-          card$append_text(comment)
-        }
-        card$append_src(source_code_r())
-        card
-      }
-      teal.reporter::simple_reporter_srv("simple_reporter", reporter = reporter, card_fun = card_fun)
-    }
-    ###
+    card_fun <- reactive({
+      req(plot_r(), plot_code_r(), setup_code_r(), libraries_code_r(), data_prep_code_r(), fit_code_r(), fitted())
+      teal.reporter::report_document(
+
+        "## Setup",
+        teal.reporter::code_chunk(setup_code_r()),
+
+        "## Libraries",
+        teal.reporter::code_chunk(libraries_code_r(), eval = TRUE),
+
+        "## Data Preparations",
+        teal.reporter::code_chunk(data_prep_code_r()),
+
+        "## Model",
+        teal.reporter::code_chunk(fit_code_r()),
+        teal.reporter::code_output(
+          paste(utils::capture.output(summary(teal.code::dev_suppress(fitted())))[-1],
+              collapse = "\n"
+          )
+        ),
+
+        "## Plot",
+        teal.reporter::code_chunk(
+          plot_code_r() |> styler::style_text() |> paste(collapse = "\n")
+        ),
+        plot_r(),
+
+        "## rtables for testing",
+        teal.reporter::code_chunk(
+          "rtables::rtable(
+          header = LETTERS[1:3],
+          rtables::rrow('one to three', 1, 2, 3),
+          rtables::rrow('more stuff', rtables::rcell(pi, format = 'xx.xx'), 'test', 'and more')
+          )"
+        ),
+        rtables::rtable(
+          header = LETTERS[1:3],
+          rtables::rrow("one to three", 1, 2, 3),
+          rtables::rrow("more stuff", rtables::rcell(pi, format = "xx.xx"), "test", "and more")
+        ),
+
+        "## Table for testing",
+        teal.reporter::code_chunk(
+          "head(iris)"
+        ),
+        head(iris),
+
+        "## keep_in_report",
+
+        "If you don't want to include code for head(mtcars) in report - just don't include it.",
+
+        "If you want the object to be kept in the report as loaded from .rds use keep_in_report()",
```

Comment:
```suggestion
        "If you want the object to be kept in the rendered output (pdf, pptx, etc.) use keep_in_report()",
```

For me high-level explaination is. If `head(mtcars) |> keep_in_report(FALSE)`:
- it will be included in the Reporter object (because it is in the list of items provided to report_document) 
- will be shown when previewing the results in the report_previewer (to avoid double evaluation of the code chunk)
- will be included in rendered in pdf, pptx, ... files directly to avoid double evaluation of the code chunk
- Will be excluded from saved .Rmd file content because .Rmd doesn't need to include a rendered output - output will be rendered when runnning .Rmd file

If `keep_in_report(TRUE)` - then always included

Reply:
Yes, your assumptions are right. I wonder if that's gonna be easy to grasp for the final user 🤔